pkg/dataobj/encoder.go

```diff
@@ -174,105 +143,105 @@ func (enc *encoder) getDictionaryKey(text string) uint32 {
 	return key
 }
 
-// MetadataSize returns an estimate of the current size of the metadata for the
-// data object. MetadataSize does not include the size of data appended or the
-// currently open stream.
-func (enc *encoder) MetadataSize() int { return proto.Size(enc.Metadata()) }
+func (enc *encoder) Metadata() (proto.Message, error) {
+	sections := make([]*filemd.SectionInfo, len(enc.sections))
+
+	offset := enc.startOffset
+
+	for i, info := range enc.sections {
```

Comment:
Seems a bit silly, but I found this much easier to follow:

```go
dataOffset := offset
metaOffset := dataOffset + info.DataSize
```

and then use these variables in the struct literal.